src/meenu/Shop.java

```diff
@@ -0,0 +1,121 @@
+package meenu;
+
+/*Create a system (Shop) which maintains each item's quantity.
+
+a. Initial stocks of Maggie: 50 Maggie packets 
+b. Initial stocks of Dosa : 43 packets 
+c. Initial stocks of pouches : 39 oil packets 
+d. Initial stocks of panipuri : 43 packets  
+e. Initial stocks of masala�s : 73 packets.
+
+Expectation:
+1) Only one Customer came for shopping, he picked different items with different quantities, print "running out of stock" message if a particular item is sold out otherwise deduct requested quantity from total quantity. Partial selling is allowed.  
+2) Create a method which shows only those item which is "Out of stock".
+3) Create a method which shows only those item which is "Available in stock" & respective quantities.
+*/
+
+public class Shop {
+	
+	int maggieQ = 50;
+	int dosaQ = 43;
+	int oilQ = 39;
+	int panipuriQ = 43;
+	int masalaQ =73;
+	boolean partialAllowed;
+	
+	void requiredQuantity(int maggie, int dosa,int oil,int panipuri,int masala,boolean pAllowed) {
+		if(maggieQ == 0)
+			System.out.println("Maggie is out of stock");
+		else if(maggie<=maggieQ) 
+		    maggieQ = maggieQ - maggie;
+		else {
+			if(pAllowed) {
+				System.out.println("Sorry,we don't have sufficent quantity of maggie. We can give you: "+maggieQ);
+				maggieQ = 0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct